views/default/elgg.js.php

```diff
@@ -51,11 +51,13 @@
 	echo "\n";
 }
 
-/**
- * Set some values that are cacheable
- */
+// Set some values that are cacheable
+$data = (object)elgg_trigger_plugin_hook('elgg.data', 'site', null, []);
 ?>
 //<script>
+// page data overrides site data
+elgg.data = $.extend(<?= json_encode($data) ?>, elgg._data);
+delete elgg._data;
 
 elgg.version = '<?php echo elgg_get_version(); ?>';
```

Comment:
Can we move these into PHP array that is passed to the hook, or merged with the hook return value? Would be nice to clean up the spaghetti code below.